### PR TITLE
Bump xml2js from 0.4.23 to 0.5.0

### PR DESCRIPTION
**Pull request by aoguan1990**
### Issues Resolved

Description:
- Bump package `xml2js` from `0.4.22` to `0.5.0` to resolve [CVE-2023-0842](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3793) in branch main

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by aoguan1990** [comment]
@joshuarrrr We still observed the Link checker failed for the two links in below. Can you please help check on that?
```
Errors in src/plugins/discover/public/application/doc_views/doc_views_helpers.tsx
✗ http://roubenmeschian.com/rubo/?p=51 (HTTP status server error (500 Internal Server Error) for url (https://roubenmeschian.com/rubo/?p=51))

Errors in src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx
✗ http://roubenmeschian.com/rubo/?p=51 (HTTP status server error ([50](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4661878096/jobs/8251639176?pr=3819#step:5:51)0 Internal Server Error) for url (https://roubenmeschian.com/rubo/?p=[51](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4661878096/jobs/8251639176?pr=3819#step:5:52)))
```

**Comment by joshuarrrr** [comment]
@aoguan1990 The link that's causing the link check failure is unreachable: http://roubenmeschian.com/rubo/?p=51

Can you open an issue to find where that link is referenced and remove it?

**Comment by aoguan1990** [comment]
> http://roubenmeschian.com/rubo/?p=51

Removed the unused url: http://roubenmeschian.com/rubo/?p=51. Thanks!

**Comment by aoguan1990** [comment]
> @aoguan1990 The link that's causing the link check failure is unreachable: http://roubenmeschian.com/rubo/?p=51
> 
> Can you open an issue to find where that link is referenced and remove it?

Removed the unused url: http://roubenmeschian.com/rubo/?p=51. Thanks! @joshuarrrr

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [CVE-2023-25166] Bump formula to 3.0.1 ([#3416](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3416))
 - [CVE-2023-25653] Bump node-jose to 2.2.0 ([#3445](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3445))
 - [CVE-2023-26486][cve-2023-26487] Bump vega from 5.22.1 to 5.23.0 ([#3533](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3533))
+- [CVE-2023-0842]Bump xml2js from 0.4.23 to 0.5.0 ([#3842](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3842))
```
Comment (joshuarrrr) [inline CHANGELOG.md:24]:
```suggestion
- [CVE-2023-0842] Bump xml2js from 0.4.23 to 0.5.0 ([#3842](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3842))
```

File: src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx [hunk 2]
```diff
@@ -88,7 +88,6 @@ export function convertDirectiveToRenderFn(
     return () => {
       if (!rejected) {
         // for cleanup
-        // http://roubenmeschian.com/rubo/?p=51
```
Comment (joshuarrrr) [inline src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx:91]: Can we open these link removal changes as a separate PR?

Comment (aoguan1990) [reply src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx:91]: Address this issue with PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3847.  Thanks!

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

File: src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx (hunk 2 above)
Comment (AMoo-Miki) [reply src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx:91]: Let's rebase this PR after #3847 is merged in.

File: CHANGELOG.md (hunk 1 above)
Comment (aoguan1990) [reply CHANGELOG.md]: Thanks for the comments, addressed as per comment.

File: yarn.lock [hunk 3]
```diff
@@ -18433,14 +18433,22 @@ xml2js@0.4.19:
     sax ">=0.6.0"
     xmlbuilder "~9.0.1"
 
-xml2js@^0.4.22, xml2js@^0.4.5:
+xml2js@^0.4.5:
   version "0.4.23"
```
[the remark this answers is not in the document]
Comment (aoguan1990) [reply yarn.lock]: Thank you so much for the help! Address this issue as per comment.

**Review by AMoo-Miki — DISMISSED** [review]

**Review by joshuarrrr — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]

**Pull request merged**